src/components/Radio.tsx

```diff
@@ -40,18 +43,36 @@ const StyledInput = styled.input`
   }
 
   &:checked::before {
-    opacity: 1;
+    opacity: ${(props => props.isDisabled ? 0 : 1)};
   }
 `;
 
+const LabelWithTooltipWrapper = styled.div`
+  display: inline-flex;
+`;
+
 type RadioProps = PropsWithChildren<
   Omit<InputHTMLAttributes<HTMLInputElement>, 'type'>>;
 
-export const Radio = ({ children, ...props }: RadioProps) => {
-  return (
-    <StyledLabel disabled={props.disabled}>
-      <StyledInput {...props} type="radio" />
-      {children}
-    </StyledLabel>
-  );
+export const Radio = ({ children, disabled, ...props }: RadioProps & {tooltipText?: string}) => {
+
+  const state = useTooltipTriggerState({delay: 0});
+  const ref = React.useRef(null);
+
+  const { triggerProps, tooltipProps } = useTooltipTrigger({delay: 0}, state, ref);
+
+  return props.tooltipText && disabled
```

Comment:
Only rendering the tooltip when disabled is surprising to me, I'd expect it to always work if the prop was sent. Could we make `tooltipText` conditional where Radio renders instead?